internal/backend/server.go

```diff
@@ -847,21 +854,142 @@ func setRealIPRequest(req *http.Request) *http.Request {
 }
 
 type server struct {
-	server *http.Server
-	certData
+	server   *http.Server
+	certData *certData
 }
 
 type certData struct {
+	cert     tls.Certificate
 	certFile string
 	keyFile  string
+	mu       sync.Mutex
+	loaded   bool
+}
+
+func (c *certData) load() error {
+	cert, err := tls.LoadX509KeyPair(c.certFile, c.keyFile)
+
+	if err == nil {
+		c.loaded = true
+	}
+
+	if err != nil {
+		return err
+	}
```

Comment:
This is a bit weird, shouldn't we just check for the error first like usual, and below simply set loaded=true?